contrib/go_plugin_api/filter.go

```diff
@@ -0,0 +1,70 @@
+package plugin
```

Comment:
I would like to see a complete working example, exercised in CI. If you have to include all the code temporarily that's fine, but I still want to understand what the developer experience looks like, and have it be documented. Thank you!

Reply:
Now we have implemented a simple version example ` <https://github.com/mosn/envoy-go-filter-samples/blob/master/simple/filter.go>` based on this API that can use go-plugins to affect request processing in Envoy.